perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -8116,10 +8117,19 @@ public void convertAttributeToUnique(PerunSession session, int attrId) throws At
 		log.info("converting attribute {} to unique",attrDef.getName());
 		attrDef.setUnique(true);
 		this.updateAttributeDefinition(session, attrDef);
-		long startTime = System.currentTimeMillis();
+
 		attributesManagerImpl.convertAttributeValuesToUnique(session, attrDef);
-		long endTime = System.currentTimeMillis();
-		log.debug("Attribute {} was converted to unique in {} ms",attrDef.getName(),(endTime-startTime));
+	}
+
+	@Override
+	public int convertAttributeToNonunique(PerunSession session, int attrId) throws AttributeNotExistsException, AttributeNotMarkedUniqueException {
+		AttributeDefinition attrDef = getAttributeDefinitionById(session, attrId);
+		if(!attrDef.isUnique()) throw new AttributeNotMarkedUniqueException("Cannot convert attribute because it is already marked as unique", attrDef);
```

Comment:
Exception message should state ``non unique`` at the end.

Reply:
Fixed.